vitrina/projects/services.py

```diff
@@ -1,46 +1,94 @@
 from vitrina.users.models import User
 from vitrina.projects.models import Project
-from vitrina.orgs.services import has_perm, is_representative, Action
-from vitrina.orgs.models import Representative, Organization
 from django.db.models import Q
-from django.contrib.contenttypes.models import ContentType
 from django.db.models import QuerySet
+from django.contrib.auth.models import AnonymousUser
 
 
-def can_update_project(user: User, project: Project) -> bool:
+def _q_visible_projects(user: User | AnonymousUser) -> Q:
+    public_approved = Q(is_public=True, status=Project.APPROVED)
+
+    if not getattr(user, "is_authenticated", False):
+        return public_approved
+
     if user.is_staff or user.is_superuser:
-        return True
+        return Q()
+
+    represented_org_ids = user.represented_org_ids
+
+    filter = (
```

Comment:
`filter` shadows built-in use. 

```suggestion
    q_filter = (
```

Reply:
Let's just not use the one-letter variable names, and do something like this instead:

```suggestion
    query_filter = (
```